src/lib/mina_base/parties.ml

```diff
@@ -274,7 +274,17 @@ let check_depths (t : t) =
     true
   with _ -> false
 
-let check (t : t) = check_depths t
+(* Fee payer MUST use a full commitment, and other parties MAY use a full 
+   commitment IFF they are also the fee payer. *)
+let check_commitment_kinds ({ fee_payer; other_parties; _ } : t) =
+  fee_payer.data.body.use_full_commitment
```

Comment:
It might make more sense to have `use_full_commitment` be `unit` for the fee payer and fill it in when converting, like we do for e.g. `token_id`. Once that is done, this function can go away.